cni/go.mod

```diff
@@ -0,0 +1,83 @@
+module github.com/Azure/azure-container-networking/cni
+
+go 1.17
+
+replace (
+	github.com/Azure/azure-container-networking => ../
+	github.com/Azure/azure-container-networking/cns => ../cns
```

Comment:
go standard says (https://github.com/golang/go/wiki/Modules#when-should-i-use-the-replace-directive)
`Note: if the right-hand side of a replace directive is a filesystem path, then the target must have a go.mod file at that location`

do we need to create go.mod under cns directory?

Reply:
this `replace` is temporary - once versions of the cni and cni module are tagged, we can remove the `replace` and instead use `require` for the specific verson of the module that we have tagged.
for example, if we tag `cns/v1.4.16` and `cni/v1.4.16`, then cni and cns can depend on those published module versions, respectively.